I joined the QCB department at USC as a PhD student in August 2019. Since then, I have been diligently
working on my research projects alongside required coursework of the program. When the COVID19
pandemic changed everything about our life during March 2020, I was very unsure of the future.
Thankfully, due to wise and loving care of the university, my adviser Prof. Remo Rohs and other members of the QCB
department, I was able to continue my work through the difficult times. I have learned much in the
past one and three quarter years, both academically and about life in academia. Prof. Adam MacLean
worked really hard to help me finish the RVAgene project and get it published in the  OUP
Bioinformatics journal. The whole process of writing up, submitting, revising the manuscript and
finally getting it accepted by the journal was an enlightening experience for me. I have also been
working with Jared Sagendorf from Rohs Lab, to finish up the Geobind project and to
get started on the proposed binding element design project.  I hope to make this project a success
in the near future. 

Since joining Rohs Lab as a PhD student, I have been encouraged to explore the field of RNA biology.
I have done multiple literature reviews on topics like protein-RNA binding, RNA structure
prediction etc. Thus I came across emerging computational works on lncRNAome (long
non-coding RNA). 
70-90\% mammalian genome get transcribed into RNA, but only 1\% of it get translated. Rest are known
as ncRNA (nonc coding RNA). lncRNAs are defined by their size range of 200 bases to 10kilobases.
For a long time lncRNAs were regarded as only transcriptional noise. But, recently, it has been
shown that they perform important important regulatory functions. Variations in lncRNA expression
has been shown to be significantly correlated with certain disease traits \citep{wapinski2011long}. 
\citet{al2019long} shows that lncRNA expression actually better explains certain cancer
classification dataset compared mRNA expression. Hence, there is a growing interest in developing
computational methods addressing various kinds of biological problems in lncRNAome.
\citet{alam2020deep} is a very recent review of Deep Learning methods recently being developed for
such tasks. Some of these works include, lncRNA-protein interaction prediction
\citep{pan2016ipminer, zhao2018bipartite, yi2018deep, zhan2019bgfe, peng2019rpiter}, lncRNA 
identification \citep{baek2018lncrnanet,yang2018lncadeep, tripathi2016deeplnc},
learning regulatory information \citep{alam2019deepcnpp, alam2019deepel}, predicting subcellular
localization of lncRNAs \citep{gudenas2018prediction},    lncRNA-miRNA interaction prediction
\citep{huang2019predicting}, lncRNA-disease association prediction \citep{hu2019deep, xuan2019dual,
al2019long, xuan2019graph} etc. Most of these methods are fairly recent and uses state of the art
deep learning architectures to solve these problems. However, one key problem for most if not all of
these methods is the fact that lncRNA expressions are highly tissue specific, much more compared to
mRNA expression. I personally find this point very interesting. And, as a possible future work, if
possible I would like to find out (possibly using some computational method) what are the factors
that result into such tissue specificity of lncrna expression. Such a project would require a
detailed dataset of tissue specificity. Luckily for us, lncRNAKB,  very recent knowledgebase
published by \citet{seifuddin2020lncrnakb}, gives detailed tissue specificity information  for a large
amount of lncRNA genes gathered across 6 different databases. Although it's probably a very
difficult biological problem, and may not even be solvable by our current computational
repertoire, I am interested in using this tissue specificity data to try and shed some light
upon the regulatory secrets giving rise to tissue specificity of lncRNA expression. My approach
would be to build a computational model which can reasonably explain this dataset and then try to
interpret that model to gain some biological insight into what is actually going on inside the cell. 
However, this is something I consider as only a possible future work, which I may get into after I finish the binding
element design project. 

Data driven modeling, especially probabilistic machine learning, has been the main theme of my
research. I am extremely grateful for being able to work with so many great minds here at USC and I
hope I can make more meaningful contributions to the collective gathering of human knowledge in
future. Thus I conclude my dissertation proposal.
